File: src/no-more-bets/models/upcoming_game.py
from typing import Annotated, Optional
from pydantic import Field
from .base_model import FrozenBaseModel


class UpcomingGame(FrozenBaseModel):
    """Represents an upcoming game/match from BetClick."""
    
    date: Annotated[str, Field(..., description="Match date header (e.g., 'Sob. 17/01')")]
    home_team: Annotated[str, Field(..., description="Home team name")]
    away_team: Annotated[str, Field(..., description="Away team name")]
    time: Annotated[str, Field(..., description="Match start time (e.g., '13:30')")]
    home_odds: Annotated[Optional[float], Field(None, description="Home team win odds")]
    draw_odds: Annotated[Optional[float], Field(None, description="Draw odds")]
    away_odds: Annotated[Optional[float], Field(None, description="Away team win odds")]
    url: Annotated[str, Field(..., description="BetClick URL for the match")]

File: src/NoMoreBets.Application/Common/Dto/Betting/UpcomingGame.cs
namespace NoMoreBets.Application.Common.Dto.Betting;

/// <summary>
/// Represents an upcoming game/match from Betclic Premier League page.
/// </summary>
public record UpcomingGame
{
  public required DateTime Date { get; init; }
  public required string HomeTeam { get; init; }
  public required string AwayTeam { get; init; }
  public required string Time { get; init; }
  public required string Url { get; init; }
}
